### PR TITLE
[refactor] DependencyDefinition renames

### PR DESCRIPTION
**Pull request by smackesey**
### Summary & Motivation

- Rename `DynamicDependencyDefinition.solid_name` -> `node_name`
- Delete `solid` arg of `DependencyDefinition` (update refs to use existing `node` arg)

### How I Tested These Changes

BK

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12189** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12189" target="_blank">Graphite</a> 
    * **PR #12330** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12330" target="_blank">Graphite</a> 
      * **PR #12338** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12338" target="_blank">Graphite</a>  👈
      * **PR #12337** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12337" target="_blank">Graphite</a> 
      * **PR #12336** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12336" target="_blank">Graphite</a> 
      * **PR #12335** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12335" target="_blank">Graphite</a> 
      * **PR #12331** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12331" target="_blank">Graphite</a> 
        * **PR #12332** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12332" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12338?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/definitions/dependency.py [hunk 1]
```diff
@@ -673,33 +673,18 @@ def the_job():
...
         output (Optional[str]): The name of the output that is depended on. (default: "result")
         description (Optional[str]): Human-readable description of this dependency.
-        node (str): The name of the node (op or graph) that is depended on, that is, from which the value
-            passed between the two nodes originates.
     """
 
     def __new__(
         cls,
-        solid: Optional[str] = None,
+        node: str,
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/definitions/dependency.py:684]: this is a top level exported API so I think this breaking change needs to wait for `1.2`

**Review by alangenfeld — CHANGES_REQUESTED** [review]

File: python_modules/dagster/dagster/_core/definitions/dependency.py (hunk 1 above)
Comment (smackesey) [reply python_modules/dagster/dagster/_core/definitions/dependency.py:684]: I believe we'd previously decided with 1.0 that lingering `solid` or `pipeline` args on public symbols could be deleted at any time, but perhaps I'm misremembering-- @sryza what do you think

Comment (sryza) [reply python_modules/dagster/dagster/_core/definitions/dependency.py:684]: I agree with @smackesey on this. @alangenfeld - feel free to push back though.

Comment (alangenfeld) [reply python_modules/dagster/dagster/_core/definitions/dependency.py:684]: My main concern here is that the docs example that I expect people may have copy pasted to start from was expclitly using `solid=` up until this PR so I think has a higher chance of breaking people.

File: docs/content/concepts/ops-jobs-graphs/graphs.mdx [hunk 2]
```diff
@@ -346,7 +346,7 @@ def construct_graph_with_yaml(yaml_file, op_defs) -> GraphDefinition:
         op_deps_entry = {}
         for input_name, input_data in op_yaml_data.get("deps", {}).items():
             op_deps_entry[input_name] = DependencyDefinition(
-                solid=input_data["op"], output=input_data.get("output", "result")
```
Comment (alangenfeld) [inline docs/content/concepts/ops-jobs-graphs/graphs.mdx:349]: this specifically

File: python_modules/dagster/dagster/_core/definitions/dependency.py (hunk 1 above)
Comment (sryza) [reply python_modules/dagster/dagster/_core/definitions/dependency.py:684]: that sounds reasonable to me

Comment (smackesey) [reply python_modules/dagster/dagster/_core/definitions/dependency.py]: >I think we should update the docs and swap node in to first position in this PR and then stack a 1.2 PR to remove solid from the constructor.

done

**Review by alangenfeld — APPROVED** [review]
👍

**Pull request merged**